### ISSUE TITLE
GPT-2 models generate punctuation instead of text

### ISSUE THREAD

**Issue by tdoublep** [torch-spyre/spyre-inference#634]
> Auto-generated by Claude (Claude Code), from a sweep of the top 100 most-downloaded HF models on Spyre. Numbers below are from real runs on `main` @ 6faf60e / torch-spyre `a312898`, compiled (`STOCK_TORCH_COMPILE`), float16.

Every `GPT2LMHeadModel` in the top 100 generates punctuation instead of text. Prompt is `"The capital of France is"`, greedy, 5 tokens, compared against stock transformers on CPU in float16:

| Model | CPU reference | Spyre |
|---|---|---|
| `openai-community/gpt2` | `' the capital of the French'` | `'-----'` |
| `distilbert/distilgpt2` | `' the capital of the French'` | `',....'` |
| `openai-community/gpt2-large` | `' the capital of France.'` | `' the,,,,'` |

## Not numerical noise

At the first differing token, Spyre is confidently predicting punctuation while the reference predicts content words. For `gpt2` the top-5 sets do not overlap at all:

```
ref top-5:  262 ' the' -2.44 | 783 ' now' -3.07 | 257 ' a' -3.07 | 6342 ' Paris' -3.44 | 4881 ' France' -3.44
spy top-5:  12  '-'    -0.90 | 11  ','    -2.53 | 13  '.'   -2.65 | 986  '...'   -2.90 | 198  '\n'    -3.03
```

Spyre puts ~40% on `'-'`. Same shape for the other two. This also fails upstream vLLM's tolerant `check_logprobs_close` criterion (each side's token in the other's top-5), so it is not a float16 tie.

`head_dim` is 64 on all three, so this is not the head-size padding path.

## A possible lead

GPT-2 is the one architecture here whose HF weights are stored **transposed**: it uses `Conv1D` (`y = x @ W + b`, `W` shaped `[in, out]`) everywhere a decoder would use `nn.Linear`. vLLM handles this in `gpt2.py::_transpose_conv1d`, which does `loaded_weight = loaded_weight.t()` for `c_attn` / `c_proj` / `c_fc`.

`.t()` returns a non-contiguous view. If the Spyre optimal-Linear layout conversion assumes a contiguous weight, these three models would be the only ones in the top 100 to hit that. Worth checking, though I have not tested it.

For reference, hf-adapters takes a different route: `hf_gpt2.py::prepare_for_spyre` rewrites every `Conv1D` into a real `nn.Linear` and splits the fused `c_attn` into separate q/k/v before handing anything to Spyre.

## Possibly related

`bigscience/bloomz-560m` fails the same way (Spyre's confident pick is absent from the reference's top-5). GPT-2 and BLOOM both have a **fused** QKV projection; `facebook/opt-125m`, which is also a non-RoPE decoder but has separate q/k/v, passes 5/5. Might be one bug or two.
